pkg/async/schedule/flytescheduler/workflow_executor.go

```diff
@@ -0,0 +1,234 @@
+package flytescheduler
+
+import (
+	"context"
+	"fmt"
+	flyteSchedulerInterfaces "github.com/flyteorg/flyteadmin/pkg/async/schedule/flytescheduler/interfaces"
+	"github.com/flyteorg/flyteadmin/pkg/async/schedule/interfaces"
+	mgInterfaces "github.com/flyteorg/flyteadmin/pkg/manager/interfaces"
+	"github.com/flyteorg/flyteadmin/pkg/repositories"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
+	runtimeInterfaces "github.com/flyteorg/flyteadmin/pkg/runtime/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/google/uuid"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"os"
+	"strings"
+	"time"
+)
+
+const executorSleepTime = 30
+
+const snapshotWriterSleepTime = 30
+const backOffSleepTime = 60
+
+type workflowExecutor struct {
+	db                   repositories.RepositoryInterface
+	config               runtimeInterfaces.Configuration
+	executionManager     mgInterfaces.ExecutionInterface
+	snapshot             flyteSchedulerInterfaces.Snapshot
+	snapShotReaderWriter flyteSchedulerInterfaces.SnapshotReaderWriter
+	goGfInterface        flyteSchedulerInterfaces.GoGFWrapper
+}
+
+func (w *workflowExecutor) CheckPointState(ctx context.Context) {
+	workflowExecConfig := w.config.ApplicationConfiguration().GetSchedulerConfig().WorkflowExecutorConfig.FlyteWorkflowExecutorConfig
+	fileName := workflowExecConfig.SnapshotFileName
+	for true {
+		f, err := os.OpenFile(fileName, os.O_WRONLY | os.O_CREATE, 0644)
```

Comment:
why are we using a file to store state? this won't be recoverable if we bring up a new instance of a scheduler pod, right? can we use the database to store our marker?

Reply:
Makes sense. I was assuming we would have access to persistent volumes to our pods. If thats not the case then yes storing it the DB would be right thing to do for now.
I will update on this

Reply:
Done